src/Sarif.Viewer.VisualStudio/SdkUiUtilities.cs

```diff
@@ -1080,5 +1086,82 @@ internal static bool IsShellInCommandLineMode(System.IServiceProvider servicePro
+        internal static List<Inline> GetInlinesForErrorMessage(string message)
+        {
+            return GetInlinesForErrorMessage(message, null, null);
+        }
+
+        /// <summary>
+        /// Builds a set of Inline elements from the specified message, optionally with embedded hyperlinks.
+        /// </summary>
+        /// <param name="message">The message to process.</param>
+        /// <param name="createHyperlinks">True to generate embedded hyperlinks; otherwise, false.</param>
+        /// <param name="data">A data object to include in each hyperlink's Tag property.</param>
+        /// <param name="clickHandler">A delegate for the Hyperlink.Click event.</param>
+        /// <returns>A collection of Inline elements that represent the specified message.</returns>
+        internal static List<Inline> GetInlinesForErrorMessage(string message, object data, RoutedEventHandler clickHandler)
+        {
+            var inlines = new List<Inline>();
+
+            MatchCollection matches = Regex.Matches(message, s_embeddedLinkRegexPattern, RegexOptions.Compiled | RegexOptions.CultureInvariant);
+            int start = 0;
+
+            if (matches.Count > 0)
+            {
+                Group group = null;
+
+                foreach (Match match in matches)
+                {
+                    group = match.Groups["link"];
+
+                    // Add the plain text segment between the end of the last group and the current link
+                    inlines.Add(new Run(UnescapeBrackets(message.Substring(start, group.Index - 1 - start))));
+
+                    if (clickHandler != null)
+                    {
+                        var link = new Hyperlink();
+
+                        // Stash the data object and relative link id
+                        link.Tag = new Tuple<object, int>(data, Convert.ToInt32(match.Groups["index"].Value));
+
+                        // Set the hyperlink text
+                        link.Inlines.Add(new Run($"{group.Value}"));
+
+                        inlines.Add(link);
+                    }
+                    else
+                    {
+                        // Add the link text as plain text
+                        inlines.Add(new Run($"{group.Value}"));
+                    }
+
+                    start = match.Index + match.Length;
+                }
+            }
+
+            if (start < message.Length)
+            {
+                // Add the plain text segment after the last link
+                inlines.Add(new Run(UnescapeBrackets(message.Substring(start))));
+            }
+
+            return inlines;
+        }
+
+        /// <summary>
+        /// Removes escape backslashes that were used to suppress embedded linking.
+        /// </summary>
+        /// <param name="s">The string to be processed.</param>
+        /// <returns></returns>
+        internal static string UnescapeBrackets(string s)
```

Comment:
As I said in a comment on the tests, I don't think you need to test this alone, so this can be private.

Reply:
Removed